src/adapter_wrap.cc

```diff
@@ -174,3 +213,35 @@ int AdapterWrap::ListenerCount(Handle<Object> argThis, const char *eventName) {
 
   return listenersArr->Length();
 }
+
+Handle<Value> AdapterWrap::GetDevicePowerStatus(const Arguments& args) {
+    HandleScope scope;
+    
+    AdapterWrap* obj = ObjectWrap::Unwrap<AdapterWrap>(args.This());
+    CEC::cec_power_status ret;
+    if (args.Length() && args[0]->IsNumber()) {
+        ret = obj->cec_adapter->GetDevicePowerStatus((CEC::cec_logical_address) args[0]->IntegerValue());
```

Comment:
should probably validate that the values are within the valid range, ie 0-15 (-1 should not be allowed) for completeness.